src/core/comparable.ml

```diff
@@ -5,3 +5,9 @@ module type S = sig
 
   val compare:t Comparator.t
 end
+
+module type C = sig
```

Comment:
Here is an example of an implicit module signature for a parameterized type from the modular implicit paper:

```
implicit module Show_list { A : Show } = struct
type t = A . t list
let show l = "["^ String . concat ", " ( List . map A . show l ) ^"]"
end
```

The paper is here: http://www.lpw25.net/ml2014.pdf

For the most part, I think we should avoid defining any interfaces that result in unwieldy apis that will be resolved in a consistent fashion by using modular implicits.